src/main/kotlin/com/exactpro/th2/check1/CollectorService.kt

```diff
@@ -141,10 +141,14 @@ class CollectorService(
 
     private fun MessageFilter.toRootMessageFilter(): RootMessageFilter {
         return RootMessageFilter.newBuilder()
-                .setMessageType(this.messageType)
-                .setComparisonSettings(this.comparisonSettings.toRootComparisonSettings())
-                .setMessageFilter(this)
-                .build()
+            .setMessageType(this.messageType).also {
+                if (this.messageType.isNotEmpty()) {
```

Comment:
I think we should report it as an exception. We cannot continue the rule execution because we don't have a message type to check.
The user should know that the rule's parameters are incorrect as soon as possible